### DISCUSSION TITLE
Unable to catch an error when creating an AudioWorklet

### DISCUSSION THREAD

**Discussion by JSmithOner** [WebAudio/web-audio-api#2543] [Q&A] [answered]
I have an audio worklet that looks like this:

```
class CustomProcessor extends AudioWorkletProcessor {
  constructor() {
      super();
      // Create a reverb node
      this.reverbNode = new ReverbNode();
      // Connect the reverb node to the output
      this.reverbNode.connect(this.output);
    }

  process(inputs, outputs, parameters) {
    // Get the input and output
    const input = inputs[0];
    const output = outputs[0];

    // Loop through the input and output
    for (let channel = 0; channel < input.length; ++channel) {
      const inputChannel = input[channel];
      const outputChannel = output[channel];

      // Connect the input to the reverb node
      this.reverbNode.connect(inputChannel)

      // Loop through the samples
      for (let i = 0; i < inputChannel.length; ++i) {
        // Apply the reverb effect
        outputChannel[i] = this.reverbNode.process(inputChannel[i]);
      }
    }

    return true;
  }
}

registerProcessor('custom-processor', CustomProcessor);
```

This processor should not build on purpose.

So I get an expected error 

```
Uncaught ReferenceError: ReverbNode is not defined
    at new CustomProcessor (fb60f14c-281b-4ba1-bec8-411400e34a80:5:27)
```

The problem now is I've tried adding `try...catch` around the code also I've turned my code in a usual promise with `then()...catch(e)...` but the error is still uncaught and I don't don't know how to catch it.

The creation process looks a bit like that

```
try {
  const audioWorkletProcessorBlob = new Blob([ processorCode ], { type: 'application/javascript' });
  await this.audioContext.audioWorklet.addModule(URL.createObjectURL(audioWorkletProcessorBlob))
  //I've tried the previous line with then...catch(e)
  this.audioWorkletNode = new AudioWorkletNode( this.audioContext, 'custom-processor' );
} catch(e) {}
```
**UPDATE:**

I was able to catch one level of error using `this.audioWorkletNode.onprocessorerror` but the error I get is `an error thrown from AudioWorkletProcessor constructor` which is better than nothing but the other error is still uncaught and I am not able at this point to catch it to get particular details of the error.

Thanks in advance

**Comment by padenot** [comment, marked as answer]
This repository is home to the Web Audio API specification.

Questions on how to do something with the Web Audio API are better suited to either [Stack Overflow](https://stackoverflow.com/questions/tagged/web-audio) or the [Web Audio API slack](https://web-audio-slackin.herokuapp.com/).

That said, this is an implementation issue, here's what Firefox says when logging the argument of `onprocessorerror`:

![Screenshot from 2023-03-20 11-21-05](https://user-images.githubusercontent.com/270485/226312282-1a34ec41-7233-44c3-b0a2-ce149df27491.png)

It has the correct message, line and column number, filename (here it's a blob that's why it's an uuid).

You can alternatively add a `try {} catch (...) {}` in the body of the ctor, that would also work.

Please file an issue on the browser vendor's issue tracker.

- **WebKit** (e.g. Safari): https://bugs.webkit.org/enter_bug.cgi?product=WebKit&component=Web%20Audio (WebKit bugzilla account needed).
- **Blink** (e.g. Chrome, Edge): https://new.crbug.com/ (Google account needed).


For reference, **Gecko**'s (Firefox) issue tracker is at: https://bugzilla.mozilla.org/enter_bug.cgi?product=Core&component=Web%20Audio (GitHub or Mozilla Bugzilla account needed).

**Comment by JSmithOner** [comment]
Sorry I though it was the correct place. I guess I'll make a bug request and for the moment add different implementations based on browser. Best and Thanks again.
